### ISSUE TITLE
Ops agent policy > neither plan nor apply work due to local variable skip_download 

### ISSUE THREAD

**Issue by nika-pr** [terraform-google-modules/terraform-google-cloud-operations#65]
### TL;DR

Creating an Ops agent policy using terraform fails due to the variable local.skip_download which is being used in the "random_id" "cache" resource.

### Expected behavior

Following the sample from the reference (https://registry.terraform.io/modules/terraform-google-modules/cloud-operations/google/latest/submodules/agent-policy), I am trying to establish an agent policy that installs the Ops Agent on all my Debian 11 VMs.

I would expect plan and apply to work when my code is so similar to the sample.

### Observed behavior

Both plan as well as apply throw the following error:

 ```
Error: Invalid count argument
│
│   on .terraform/modules/agent_policy.gcloud-destroy/main.tf line 57, in resource "random_id" "cache":
│   57:   count = (!local.skip_download) ? 1 : 0
│
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work    
│ around this, use the -target argument to first apply only the resources that the count depends on.
╵
╷
│ Error: Invalid count argument
│
│   on .terraform/modules/agent_policy.gcloud-upsert/main.tf line 57, in resource "random_id" "cache":
│   57:   count = (!local.skip_download) ? 1 : 0
│
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work    
│ around this, use the -target argument to first apply only the resources that the count depends on.
```
 

### Terraform Configuration

```hcl
variable "GCP_BILLING_ACCOUNT_ID" {
  type = string
}

variable "GCP_PROJECT_ID" {
  type = string
}

variable "GCP_SERVICES" {
  type    = list(string)
  default = [
    "logging.googleapis.com",
    "monitoring.googleapis.com",
    "osconfig.googleapis.com"
  ]
}

resource "google_project" "worker_project" {
  billing_account = var.GCP_BILLING_ACCOUNT_ID
  name            = var.GCP_PROJECT_ID
  project_id      = var.GCP_PROJECT_ID
}

resource "google_project_service" "google_project_services" {
  project                    = google_project.worker_project.project_id
  for_each                   = toset(var.GCP_SERVICES)
  service                    = each.value
  disable_dependent_services = true
}

module "agent_policy" {
  source  = "terraform-google-modules/cloud-operations/google//modules/agent-policy"
  version = "~> 0.2.4"

  depends_on  = [google_project_service.google_project_services]
  project_id  = var.GCP_PROJECT_ID
  policy_id   = "ops-agent-policy"
  agent_rules = [
    {
      type               = "ops-agent"
      version            = "current-major"
      package_state      = "installed"
      enable_autoupgrade = true
    }
  ]
  os_types = [
    {
      short_name = "debian"
      version    = "11"
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.4.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/google v4.66.0
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/template v2.2.0
```
